narwhal/primary/src/core.rs

```diff
@@ -604,14 +611,6 @@ impl Core {
                 received: certificate.epoch()
             }
         );
-        ensure!(
-            self.gc_round < certificate.round(),
-            DagError::TooOld(
-                certificate.digest().into(),
-                certificate.round(),
-                self.gc_round
-            )
-        );
```

Comment:
For efficiency, since the cert has been voted I don't think the additional storage cost is significant. The processing cost should be the same for old and new certs, within the same category of fresh or already processed certs.

Correctness wise, maybe I'm missing things but I have not found an argument that ignoring certificates older than gc bound is valid. It seems possible to have an authority that produces certs but never became leader or has its new certs included in the dag. And when they do get included in the dag, the oldest cert is past the gc boundary on some other nodes. And if it is correct to ignore certs older than gc bound, can older certs be deleted from store as well?

Either way, I believe it is simpler to reason about and not very costly to process and store certs regardless of gc bound, and only apply gc to in-memory data.

Reply:
> And if it is correct to ignore certs older than gc bound, can older certs be deleted from store as well?

We could delete certificates from store past gc round, only if we had full nodes that would keep the whole history in order to let other nodes (slow nodes - or bootstrapping nodes) catching up. If we start deleting stuff now from our storage then we won't be able to serve nodes that try to perform their causal completion part.

Reply:
Thanks for the info. I read the code a bit more and as @gdanezis mentioned today, during consensus DAG construction GC depth limits the depth of depth-first search. In this case dropping certificates beyond GC seems OK. Will rework this logic.

Reply:
Added back GC logic, but only keeping them in the sanitize functions.